packages/langium/src/grammar/grammar-util.ts

```diff
@@ -398,12 +397,7 @@ export function loadGrammar(json: string): ast.Grammar {
         throw new Error('Could not load grammar from specified json input.');
     }
     const grammar = astNode as Mutable<ast.Grammar>;
-    const textDocument = TextDocument.create('memory://grammar.langium', 'langium', 0, '');
-    const document = documentFromText(textDocument, {
-        lexerErrors: [],
-        parserErrors: [],
-        value: grammar
-    });
+    const document = services.shared.workspace.LangiumDocumentFactory.fromModel(grammar, URI.parse('memory://grammar.langium'));
```

Comment:
What is 'memory://grammar.langium'? What is its purpose? Is this the grammar address in VSCode?

Another thought: Looks like a magic string. If we have that string in multiple places it becomes hard to change them, when necessary. A shared constant could be a solution.

Reply:
It has no real purpose, it actually makes this "file" non-addressable, by the rest of the system. The idea is that we load a model of the user-grammar during runtime to operate on it for parsing and LSP services. However, since everything operates on `LangiumDocument` objects, we need to wrap the grammar in a document and give it a bogus URI.